### PR TITLE
chore: add human readable strings to anonymized IDs

### PR DESCRIPTION
**Pull request by eysi09**
**What this PR does / why we need it**:

This PR introduces a v2 version of anonymized analytics properties. This
version includes a human readable string in the ID so that they are
easier to recognize.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: core/src/analytics/analytics.ts [hunk 1]
```diff
@@ -249,6 +273,20 @@ export class AnalyticsHandler {
...
+   * This readable string consists of two adjectives, followed by a noun.
+   *
+   * Also truncates the hash part to 32 char (from 128).
+   *
+   * Example: dysfunctional-graceful-request_433c84996726070996f369dfc00dd202
+   */
+  public hashV2(val: string) {
+    const readable = codenamize({ seed: val, adjectiveCount: 2 })
+    const hash = this.hash(val).slice(0, 32)
+    return `${readable}_${hash}`
```
Comment (eysi09) [inline core/src/analytics/analytics.ts:287]: I picked underscore as a delimiter since codenamize uses hyphens.

**Comment by edvald** [comment]
Feature request: Do something similar for Garden version hashes :)

**Comment by eysi09** [comment]
@edvald, I updated the PR to also add a readable user ID as "trait" on the user: https://github.com/garden-io/garden/pull/2440/files#diff-8278b2fcc5bc5afb7cf477c1b2067e9861738d0e44d81bc1334ab3b3e849d299R264

File: core/src/analytics/analytics.ts [hunk 2]
```diff
@@ -294,9 +335,13 @@ export class AnalyticsHandler {
   private getBasicAnalyticsProperties(): AnalyticsEventProperties {
     return {
       projectId: this.projectId,
+      projectIdV2: this.projectIdV2,
       projectName: this.projectName,
+      projectNameV2: this.projectNameV2,
       enterpriseProjectId: this.enterpriseProjectId,
+      enterpriseProjectIdV2: this.enterpriseProjectIdV2,
```
Comment (edvald) [inline core/src/analytics/analytics.ts:342]: Do we still want to use the "enterprise" prefix here, since we're changing the field anyway?

**Review by edvald — APPROVED** [review]
Had one (open-ended) question, but LGTM 👍

File: core/src/analytics/analytics.ts (hunk 2 above)
Comment (eysi09) [reply core/src/analytics/analytics.ts:342]: Yeah, that's a good question. Don't have a strong opinion. Would we then go for `cloudProjectId` or `cloudProjectIdV2`?

Comment (edvald) [reply core/src/analytics/analytics.ts:342]: I'd say V2 just for consistency, even if it is technically redundant.

Comment (edvald) [reply core/src/analytics/analytics.ts:342]: @eysi09 reminder on ☝️

**Pull request merged**